src/gui/mainwindow.cpp

```diff
@@ -126,6 +117,7 @@ MainWindow::MainWindow(QWidget *parent)
 #if defined(Q_OS_WIN) || defined(Q_OS_MAC)
     , m_wasUpdateCheckEnabled(false)
 #endif
+    , m_searchEngine(new SearchEngine)
```

Comment:
Oh, that's great!.. MainWindow owns SearchEngine. Moreover, it still creates SearchEngine instance regardless of whether you need it or not. It is really a good design decision!
You love optimisation so much. Somehow I don't see it here.

P.S. Sorry for my sarcasm.